src/System.Text.Json/src/System/Text/Json/Serialization/Converters/DefaultConverters.cs

```diff
@@ -0,0 +1,229 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Linq;
+using System.Reflection;
+using System.Text.Json.Serialization.Policies;
+
+namespace System.Text.Json.Serialization.Converters
+{
+    internal static class DefaultConverters
+    {
+        public static readonly JsonArrayConverterAttribute s_arrayConverterAttibute = new JsonArrayConverterAttribute();
+        public static readonly JsonEnumConverterAttribute s_enumConverterAttibute = new JsonEnumConverterAttribute();
+
+        public static Type GetTypeConverter(Type classType, JsonSerializerOptions options)
+        {
+            JsonClassAttribute attr = options.GetAttributes<JsonClassAttribute>(classType, inherit: true).FirstOrDefault();
+
+            if (attr == null)
+            {
+                return null;
+            }
+
+            Type converterType = attr.ConverterType;
+            return converterType;
+        }
+
+        public static TAttribute GetPolicy<TAttribute>(
+            Type parentClassType,
+            PropertyInfo propertyInfo,
+            JsonSerializerOptions options) where TAttribute : Attribute
+        {
+            TAttribute attr = null;
+            if (propertyInfo != null)
+            {
+                // Use Property first
+                attr = options.GetAttributes<TAttribute>(propertyInfo).FirstOrDefault();
+            }
+
+            if (attr == null)
+            {
+                // Then class type
+                attr = options.GetAttributes<TAttribute>(parentClassType, inherit: true).FirstOrDefault();
+
+                if (attr == null)
+                {
+                    // Then declaring assembly
+                    attr = options.GetAttributes<TAttribute>(parentClassType.Assembly).FirstOrDefault();
+
+                    if (attr == null)
+                    {
+                        // Then global
+                        attr = options.GetAttributes<TAttribute>(JsonSerializerOptions.GlobalAttributesProvider).FirstOrDefault();
+                    }
+                }
+            }
```

Comment:
This can be written more succinctly, e.g.
```C#
if (attr == null)
{
    attr =
        options.GetAttributes<TAttribute>(parentClassType, inherit: true).FirstOrDefault() ??
        options.GetAttributes<TAttribute>(parentClassType.Assembly).FirstOrDefault() ??
        options.GetAttributes<TAttribute>(JsonSerializerOptions.GlobalAttributesProvider).FirstOrDefault();
}
```

Reply:
Done. Thanks.